indra/sources/phosphoelm/api.py

```diff
@@ -0,0 +1,107 @@
+import csv
+import logging
+import requests
+from unidecode import unidecode
+
+from .processor import PhosphoElmProcessor
+
+try:
+    from bs4 import BeautifulSoup
+    has_soup = False
+except ImportError:
+    BeautifulSoup = None
+    has_soup = False
+
+logger = logging.getLogger(__file__)
+
+s3_bucket = 'bigmech'
+ppelm_s3_key = 'phosphoELM_data/phosphoELM_all_2015-04.dump'
+kinases_list_web = 'http://phospho.elm.eu.org/kinases.html'
+
+
+def process_from_dump(fname=None, delimiter='\t'):
+    """Process a phospho.ELM file dump
+
+    fname : str
+        File path to the phospho.ELM file dump. If none is provided,
+        the file will be downloaded from S3.
+    delimiter : str
+        The delimiter to use for csv.reader
+
+    Returns
+    -------
+    ppp : indra.sources.phosphoELM.PhosphoElmProcessor
+        An instance of a PhosphoElmProcessor containing the statements
+        generated from the file dump
+    """
+    if fname is None:
+        s3 = _get_s3_client()
+        s3_obj = s3.get_object(Bucket=s3_bucket, Key=ppelm_s3_key)
+        csv_reader = csv.reader(
+            s3_obj['Body'].read().decode('utf8').splitlines(True),
+            delimiter='\t'
+        )
+    else:
+        with open(fname, 'r') as f:
+            csv_reader = csv.reader(f.readlines(), delimiter=delimiter)
```

Comment:
Usually one can just do `csv.reader(f, ...)` is the `readlines()` part needed for some reason?

Reply:
Using only `f` makes `csv_reader` an object that has to be consumed within the `with-open` part, the idea is to provide `csv_reader` either from the s3 client or from the local file, but I can change it so we have `ppelm_json` be produced inside either of the if-else clauses instead.